project/recursive_sorting.py

```diff
@@ -44,10 +45,33 @@ def merge_sort_in_place(arr, l, r):
 
 # TO-DO: implement the Quick Sort function below
 def quick_sort( arr, low, high ):
-
+    
+    if low < high:
+        p = partition(arr, low, high)
+        quick_sort(arr, low, p - 1)
+        quick_sort(arr, p + 1, high)
     return arr
 
+def partition(arr, low, high):
+    pivot = arr[high]
+    i = low
+    for j in range(low, high - 1):
+        if arr[j] <= pivot:
+            temp = arr[i]
+            arr[i] = arr[j]
+            arr[j] = temp
+            i = i + 1
+    temp = arr[i]
+    arr[i] = arr[high]
+    arr[high] = temp
+    return i
 
+myArr = [1, 5, 7, 8, 10, 3, 12]
```

Comment:
I don't know how, but I tested my repo with this *exact* line. How is that possible?

Reply:
I guess I did paste my test line in the group chat, lol. I'm silly.